tests/test_types.py

```diff
@@ -21,6 +22,20 @@ def test_dataset(n, outd, ind):
     assert d.in_dim == ind
     assert d.out_dim == outd
 
+    # test combine datasets
+    x2 = 2*jnp.ones((n2, ind))
+    y2 = 2*jnp.ones((n2, outd))
+    d2 = Dataset(X=x2, y=y2)
+
+    d_combined = d + d2
```

Comment:
Does this operation commute? i.e., `d+d2 == d2+d`?

Reply:
No. This is inconsistent due to stacking. This would require sorting the dataset into order.

Reply:
Good point. We don't want to sort.